packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTInstance.mm

```diff
@@ -476,7 +476,11 @@ - (void)_loadScriptFromSource:(RCTSource *)source
   auto script = std::make_unique<NSDataBigString>(source.data);
   const auto *url = deriveSourceURL(source.url).UTF8String;
   _reactInstance->loadScript(std::move(script), url);
-  [[NSNotificationCenter defaultCenter] postNotificationName:@"RCTInstanceDidLoadBundle" object:nil];
+  [self callFunctionOnBufferedRuntimeExecutor:[](facebook::jsi::Runtime &_) {
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
why are we jumping back on the UI thread? I don't think that it is necessary: we should be able to post notification from a background thread.

What happens if we dispatch the notification from a different thread?

Reply:
@cipolleschi  I just keep the same logic as old arch, please see https://github.com/facebook/react-native/blob/0217d7e19cf10b6b2261ccd8e78f6bf02a55fed4/packages/react-native/React/CxxBridge/RCTCxxBridge.mm#L1068 . Seems it may have issue when post on Javascript thread directly.